addons/website_event/static/src/snippets/options.js

```diff
@@ -59,14 +57,12 @@ options.registry.WebsiteEvent = options.Class.extend({
         return this._super(...arguments);
     },
     /**
+     * Ensure that we get the event object id as we could be inside a sub-object of the event
+     * like an event.track
      * @private
      */
-    _getEventObject() {
-        const repr = this.ownerDocument.documentElement.dataset.mainObject;
-        const m = repr.match(/(.+)\((\d+),(.*)\)/);
-        return {
-            model: m[1],
-            id: m[2] | 0,
-        };
+    _getEventObjectId() {
+        const objectIds = this.currentWebsiteUrl.match(/(?<=-)[\d]+(?!-)(?!\w)(?![\d]+)/);
```

Comment:
Still trying to keep it simple, how about this?
```suggestion
        const objectIds = this.currentWebsiteUrl.match(/(?<=-)\d+(?![-\w\d])/);
```

Reply:
Nice one! 👍